Error when executing 'opencv_visualisation' binary file

I'm getting the following error when executing the 'opencv_visualisation' binary file. The installed binary file is strangely referencing my build directory. The command shows just this error and nothing more:

```
$ opencv_visualisation
OpenCV Error: Null pointer (NULL or empty buffer) in cvOpenFileStorage, file /home/user/abs/opencv-git/src/opencv-git/modules/core/src/persistence.cpp, line 2980
terminate called after throwing an instance of 'cv::Exception'
  what():  /home/user/abs/opencv-git/src/opencv-git/modules/core/src/persistence.cpp:2980: error: (-27) NULL or empty buffer in function cvOpenFileStorage

Aborted (core dumped)
```

- OpenCV version: 3.1.0.r781.gd06bdd5 (git master)
- Host OS: Linux (Arch Linux x86_64)
- GCC: 6.1.1
- CMake: 3.5.2
- CPU: Intel Haswell

Commands used for building:
```
$ cmake \
	-DBUILD_SHARED_LIBS:BOOL=ON \
	\
	-DBUILD_opencv_apps:BOOL=ON             -DBUILD_opencv_calib3d:BOOL=ON  	-DBUILD_opencv_core:BOOL=ON \
	-DBUILD_opencv_features2d:BOOL=ON       -DBUILD_opencv_flann:BOOL=ON    	-DBUILD_opencv_highgui:BOOL=ON \
	-DBUILD_opencv_imgcodecs:BOOL=ON        -DBUILD_opencv_imgproc:BOOL=ON  	-DBUILD_opencv_ml:BOOL=ON \
	-DBUILD_opencv_objdetect:BOOL=ON        -DBUILD_opencv_photo:BOOL=ON    	-DBUILD_opencv_shape:BOOL=ON \
	-DBUILD_opencv_stitching:BOOL=ON        -DBUILD_opencv_superres:BOOL=ON 	-DBUILD_opencv_ts:BOOL=ON \
	-DBUILD_opencv_video:BOOL=ON            -DBUILD_opencv_videoio:BOOL=ON          -DBUILD_opencv_videostab:BOOL=ON \
	-DBUILD_opencv_world:BOOL=OFF \
	\
	-DBUILD_opencv_cudaarithm:BOOL=OFF      -DBUILD_opencv_cudabgsegm:BOOL=OFF      -DBUILD_opencv_cudacodec:BOOL=OFF \
	-DBUILD_opencv_cudafeatures2d:BOOL=OFF  -DBUILD_opencv_cudafilters:BOOL=OFF     -DBUILD_opencv_cudaimgproc:BOOL=OFF \
	-DBUILD_opencv_cudalegacy:BOOL=OFF      -DBUILD_opencv_cudaobjdetect:BOOL=OFF   -DBUILD_opencv_cudaoptflow:BOOL=OFF \
	-DBUILD_opencv_cudastereo:BOOL=OFF      -DBUILD_opencv_cudawarping:BOOL=OFF     -DBUILD_opencv_cudev:BOOL=OFF \
	\
	-DCUDA_64_BIT_DEVICE_CODE:BOOL=OFF \
	-DCUDA_FAST_MATH:BOOL=OFF \
	\
	-DCMAKE_BUILD_TYPE:STRING=Release \
	-DCMAKE_COLOR_MAKEFILE:BOOL=ON \
	-DCMAKE_INSTALL_PREFIX:PATH=/usr \
	-DCPACK_BINARY_DEB:BOOL=OFF \
	\
	-DPYTHON2_NUMPY_INCLUDE_DIRS:PATH=/usr/include/python2.7/numpy \
	-DPYTHON3_NUMPY_INCLUDE_DIRS:PATH=/usr/include/python3.5m/numpy \
	\
	-DENABLE_FAST_MATH:BOOL=ON \
	\
	-DENABLE_AVX:BOOL=OFF   -DENABLE_AVX2:BOOL=OFF   -DENABLE_FMA3:BOOL=OFF   -DENABLE_POPCNT:BOOL=OFF \
	-DENABLE_SSE:BOOL=ON    -DENABLE_SSE2:BOOL=ON    -DENABLE_SSE3:BOOL=ON    -DENABLE_SSSE3:BOOL=OFF \
    -DENABLE_SSE41:BOOL=OFF -DENABLE_SSE42:BOOL=OFF" \
	\
	-DBUILD_EXAMPLES:BOOL=OFF \
	-DBUILD_PERF_TESTS:BOOL=OFF \
	-DBUILD_TESTS:BOOL=OFF \
	-DINSTALL_C_EXAMPLES:BOOL=OFF \
	-DINSTALL_PYTHON_EXAMPLES:BOOL=OFF \
	-DINSTALL_TESTS:BOOL=OFF \
	\
	-DWITH_1394:BOOL=ON       -DWITH_CLP:BOOL=OFF            -DWITH_CUBLAS:BOOL=OFF        -DWITH_CUDA:BOOL=OFF \
	-DWITH_CUFFT:BOOL=OFF     -DWITH_EIGEN:BOOL=ON           -DWITH_FFMPEG:BOOL=OFF        -DWITH_GDAL:BOOL=OFF \
	-DWITH_GIGEAPI:BOOL=OFF   -DWITH_GPHOTO2:BOOL=ON         -DWITH_GSTREAMER:BOOL=ON      -DWITH_GSTREAMER_0_10:BOOL=OFF \
	-DWITH_GTK:BOOL=ON        -DWITH_GTK_2_X:BOOL=ON         -DWITH_IPP_A:BOOL=ON          -DWITH_IPP:BOOL=ON \
	-DWITH_JASPER:BOOL=ON     -DWITH_JPEG:BOOL=ON            -DWITH_LIBV4L:BOOL=ON         -DWITH_NVCUVID:BOOL=ON \
	-DWITH_OPENCL:BOOL=ON     -DWITH_OPENCLAMDBLAS:BOOL=ON   -DWITH_OPENCLAMDFFT:BOOL=ON   -DWITH_OPENCL_SVM:BOOL=ON \
	-DWITH_OPENEXR:BOOL=ON    -DWITH_OPENGL:BOOL=ON          -DWITH_OPENMP:BOOL=ON         -DWITH_OPENNI:BOOL=OFF \
	-DWITH_OPENNI2:BOOL=OFF   -DWITH_PNG:BOOL=ON             -DWITH_PVAPI:BOOL=OFF         -DWITH_QT:BOOL=OFF \
	-DWITH_TBB:BOOL=ON        -DWITH_TIFF:BOOL=ON            -DWITH_UNICAP:BOOL=OFF        -DWITH_V4L:BOOL=ON \
	-DWITH_VTK:BOOL=OFF       -DWITH_WEBP:BOOL=ON            -DWITH_XIMEA:BOOL=OFF         -DWITH_XINE:BOOL=OFF \
	-DWITH_MATLAB:BOOL=OFF    -DWITH_OPENCLAMDFFT:BOOL=ON    -DWITH_PTHREADS_PF:BOOL=ON    -DWITH_VA_INTEL:BOOL=ON \
	\
	-DBUILD_JASPER:BOOL=OFF   -DBUILD_JPEG:BOOL=OFF          -DBUILD_OPENEXR:BOOL=OFF      -DBUILD_PNG:BOOL=OFF \
	-DBUILD_TIFF:BOOL=OFF     -DBUILD_ZLIB:BOOL=OFF \
	\
	-DOPENCV_WARNINGS_ARE_ERRORS:BOOL=OFF \
	-Wno-dev \
	.
$ make
```

Compiler and linker options (system wide set):
```
CPPFLAGS="-D_FORTIFY_SOURCE=2"
CFLAGS="-march=native -O2 -pipe -fstack-protector-strong"
CXXFLAGS="-march=native -O2 -pipe -fstack-protector-strong"
LDFLAGS="-Wl,-O1,--sort-common,--as-needed,-z,relro"
MAKEFLAGS="-j4"
```

Any help would be appreciated.
Thank you.

--- a/apps/visualisation/opencv_visualisation.cpp
+++ b/apps/visualisation/opencv_visualisation.cpp
@@ -47,7 +47,7 @@ Software for visualising cascade classifier models trained by OpenCV and to get 
 understanding of the used features.
 
 USAGE:
-./visualise_models -model <model.xml> -image <ref.png> -data <output folder>
+./opencv_visualisation --model=<model.xml> --image=<ref.png> --data=<video output folder>
 
 LIMITS
 - Use an absolute path for the output folder to ensure the tool works
@@ -81,20 +81,23 @@ struct rect_data{
 
 int main( int argc, const char** argv )
 {
+    CommandLineParser parser(argc, argv,
+        "{ help h usage ? |      | show this message }"
+        "{ image i        |      | (required) path to reference image }"
+        "{ model m        |      | (required) path to cascade xml file }"
+        "{ data d         |      | (optional) path to video output folder }"
+    );
     // Read in the input arguments
-    string model = "";
-    string output_folder = "";
-    string image_ref = "";
-    for(int i = 1; i < argc; ++i )
-    {
-        if( !strcmp( argv[i], "-model" ) )
-        {
-            model = argv[++i];
-        }else if( !strcmp( argv[i], "-image" ) ){
-            image_ref = argv[++i];
-        }else if( !strcmp( argv[i], "-data" ) ){
-            output_folder = argv[++i];
-        }
+    if (parser.has("help")){
+        parser.printMessage();
+        return 0;
+    }
+    string model(parser.get<string>("model"));
+    string output_folder(parser.get<string>("data"));
+    string image_ref = (parser.get<string>("image"));
+    if (model.empty() || image_ref.empty()){
+        parser.printMessage();
+        return -1;
     }
 
     // Value for timing
@@ -106,8 +109,11 @@ int main( int argc, const char** argv )
 
     // Open the XML model
     FileStorage fs;
-    fs.open(model, FileStorage::READ);
-
+    bool model_ok = fs.open(model, FileStorage::READ);
+    if (!model_ok){
+        cerr << "the cascade file '" << model << "' could not be loaded." << endl;
+        return  -1;
+    }
     // Get a the required information
     // First decide which feature type we are using
     FileNode cascade = fs["cascade"];
@@ -129,6 +135,10 @@ int main( int argc, const char** argv )
     int resize_factor = 10;
     int resize_storage_factor = 10;
     Mat reference_image = imread(image_ref, IMREAD_GRAYSCALE );
+    if (reference_image.empty()){
+        cerr << "the reference image '" << image_ref << "'' could not be loaded." << endl;
+        return -1;
+    }
     Mat visualization;
     resize(reference_image, visualization, Size(reference_image.cols * resize_factor, reference_image.rows * resize_factor));
 

